torch/fx/passes/shape_prop.py

```diff
@@ -69,6 +69,59 @@ def _extract_tensor_metadata(result : torch.Tensor, include_contiguity=True) ->
     return TensorMetadata(
         shape, dtype, requires_grad, stride, memory_format, is_quantized, qparams)
 
+
+@compatibility(is_backward_compatible=True)
+class SparseTensorMetadata(NamedTuple):
+    """
+    Metadata relevant to a sparse tensor with given element dtype and shape.
+
+    layout in [ torch.sparse_coo/csr/csc/bsr/bsc ]
+    batch_dim + sparse_dim + dense_dim = ndim = len(shape)
+    pos/crd_dtype denote the types used for overhead storage
+    """
+    dtype: torch.dtype
+    shape: torch.Size
+    layout: torch.layout
+    batch_dim: int
+    sparse_dim: int
+    dense_dim: int
+    blocksize: Optional[Tuple[int,int]]
+    pos_dtype: torch.dtype
+    crd_dtype: torch.dtype
+
+def _extract_sparse_tensor_metadata(t: torch.Tensor) -> SparseTensorMetadata:
+    """
+    Extract the SparseTensorMetadata of a tensor.
+    """
+    if t.layout is torch.sparse_coo:
+        assert t.ndim == t.dense_dim() + t.sparse_dim()  # no batch dim
+        pdtp, cdtp = t.indices().dtype, t.indices().dtype
+    elif t.layout is torch.sparse_csr or t.layout is torch.sparse_bsr:
+        assert t.sparse_dim() == 2
+        pdtp, cdtp = t.crow_indices().dtype, t.col_indices().dtype
+    elif t.layout is torch.sparse_csc or t.layout is torch.sparse_bsc:
+        assert t.sparse_dim() == 2
+        pdtp, cdtp =  t.ccol_indices().dtype, t.row_indices().dtype
+    else:
+        raise RuntimeError(f"Unsupported sparse layout for {t}")
+
+    if t.layout is torch.sparse_bsr or t.layout is torch.sparse_bsc:
+      bsz = t.values().shape[1:3]
```

Comment:
Taking into account batch dimensions, this should read:
```suggestion
      batch_dim = t.ndim - t.dense_dim() - t.sparse_dim()
      bsz = t.values().shape[batch_dim + 1:batch_dim + 3]
```

Reply:
Also, I suggest renaming `bsz` to `blocksize` for the consistency of the rest of the code.

Reply:
Good points